autotest/gcore/mask.py

```diff
@@ -1170,6 +1170,40 @@ def mask_26():
 # Cleanup.
 
 
+###############################################################################
+# Extensive test of nodata mask for all complex types using real part only
+
+
+def mask_27():
+
+    types = [gdal.GDT_CFloat32, gdal.GDT_CFloat64]
+
+    nodatavalue = [0.5, 0.5]
+
+    drv = gdal.GetDriverByName('GTiff')
+    for i, typ in enumerate(types):
+        ds = drv.Create('tmp/mask27.tif', 1, 1, 1, typ)
+        ds.GetRasterBand(1).Fill(nodatavalue[i] + 10j)
```

Comment:
this is invalid. must be rewritten as ds.GetRasterBand(1).Fill(nodatavalue[i], 10)